[BUG] OCR is not executed since upgrade to v3.0

### Description

Hello,

after upgrading to paperless 3.0, documents from my scanner are not OCR'ed anymore with the new default settings. Worked like a charm before. I suspect something to do with the "OCR and Archive File Generation Settings" mentioned in the migration log.

The issue is that my scanner does not perform OCR, still paperless seems to think so. Also, it interprets the PDF to be "born-digital" which is not the case.
"Mode" and "Archive File Generation" in my OCR settings are blank.

```
[2026-07-27 19:06:45,571] [DEBUG] [paperless.consumer] [4ce923d5] Archive: no — born-digital PDF (structure tags detected), ARCHIVE_FILE_GENERATION=auto

[2026-07-27 19:06:45,594] [DEBUG] [paperless.parsing.tesseract] [4ce923d5] Text detection: original_has_text=True (text_length=0, mode=auto, produce_archive=False)

[2026-07-27 19:06:45,594] [DEBUG] [paperless.parsing.tesseract] [4ce923d5] Document has text and no archive requested; skipping OCRmyPDF entirely.
```

I have attached the PDF in question to the issue.

[Test.pdf](https://github.com/user-attachments/files/30430273/Test.pdf)

With mode set to force, reprocessing the PDF still(!) yields no OCR for the file:
```
[2026-07-27 19:15:19,916] [DEBUG] [paperless.consumer] Archive: no — born-digital PDF (structure tags detected), ARCHIVE_FILE_GENERATION=auto

[2026-07-27 19:15:19,932] [INFO] [paperless.parsing.tesseract] pdftotext exited 0

[2026-07-27 19:15:19,934] [DEBUG] [paperless.parsing.tesseract] Text detection: original_has_text=True (text_length=0, mode=force, produce_archive=False)

[2026-07-27 19:15:19,934] [DEBUG] [paperless.parsing.tesseract] OCR strategy: full OCR — OCR_MODE=force

[2026-07-27 19:15:19,934] [DEBUG] [paperless.parsing.tesseract] Calling OCRmyPDF with args: {'input_file_or_options': PosixPath('/usr/src/paperless/media/documents/originals/0000207.pdf'), 'output_file': PosixPath('/tmp/paperless/paperless-v5w8og6f/archive.pdf'), 'use_threads': True, 'jobs': 8, 'language': 'deu+eng', 'output_type': 'pdfa', 'progress_bar': False, 'color_conversion_strategy': ColorConvertChoices.RGB, 'force_ocr': True, 'clean': True, 'deskew': True, 'rotate_pages': True, 'rotate_pages_threshold': 12.0, 'sidecar': PosixPath('/tmp/paperless/paperless-v5w8og6f/sidecar.txt')}

[2026-07-27 19:15:20,541] [WARNING] [ocrmypdf._pipeline] This PDF is marked as a Tagged PDF. This often indicates that the PDF was generated from an office document and does not need OCR. PDF pages processed by OCRmyPDF may not be tagged correctly.

[2026-07-27 19:15:21,448] [INFO] [ocrmypdf._exec.tesseract] [tesseract] Too few characters. Skipping this page

[2026-07-27 19:15:21,448] [INFO] [ocrmypdf._exec.tesseract] [tesseract] Too few characters. Skipping this page

[2026-07-27 19:15:21,448] [ERROR] [ocrmypdf._exec.tesseract] [tesseract] Error during processing.

[2026-07-27 19:15:21,448] [INFO] [ocrmypdf._pipeline] page is facing ⇧, confidence 0.00 - no change

[2026-07-27 19:15:23,906] [INFO] [ocrmypdf._graft] Parsing 1 pages with HocrParser

[2026-07-27 19:15:25,169] [INFO] [ocrmypdf._pipelines.ocr] Postprocessing...

[2026-07-27 19:15:25,557] [INFO] [ocrmypdf._pipeline] Image optimization ratio: 1.00 savings: 0.1%

[2026-07-27 19:15:25,558] [INFO] [ocrmypdf._pipeline] Total file size ratio: 0.18 savings: -447.2%

[2026-07-27 19:15:25,559] [INFO] [ocrmypdf._pipelines._common] Output file is a PDF/A-2b (as expected)
```

In paperless 2 I had no such issues and I have successfully extracted text from my documents.

I'm happy to test any bugfixes and to provide further documents, if needed. My scanner is a Epson WF-3720.
As nobody so far seems to have encountered this issue, it could just as well be an oversight on my part. If you spot something, please let me know :)

I obtained the docker image from `ghcr.io/paperless-ngx/paperless-ngx:latest`.




### Steps to reproduce

1. Import Test.pdf .
2. In the processed file, go to "Content".
3. Content is empty whereas it should have been populated by OCR:

### Webserver logs

```bash
[2026-07-27 19:06:45,561] [INFO] [paperless.consumer] [4ce923d5] Consuming Test.pdf

[2026-07-27 19:06:45,563] [DEBUG] [paperless.consumer] [4ce923d5] Detected mime type: application/pdf

[2026-07-27 19:06:45,565] [DEBUG] [paperless.consumer] [4ce923d5] Parser: Paperless-ngx Tesseract OCR Parser v3.0.2

[2026-07-27 19:06:45,568] [DEBUG] [paperless.consumer] [4ce923d5] Parsing Test.pdf...

[2026-07-27 19:06:45,571] [DEBUG] [paperless.consumer] [4ce923d5] Archive: no — born-digital PDF (structure tags detected), ARCHIVE_FILE_GENERATION=auto

[2026-07-27 19:06:45,593] [INFO] [paperless.parsing.tesseract] [4ce923d5] pdftotext exited 0

[2026-07-27 19:06:45,594] [DEBUG] [paperless.parsing.tesseract] [4ce923d5] Text detection: original_has_text=True (text_length=0, mode=auto, produce_archive=False)

[2026-07-27 19:06:45,594] [DEBUG] [paperless.parsing.tesseract] [4ce923d5] Document has text and no archive requested; skipping OCRmyPDF entirely.

[2026-07-27 19:06:45,594] [DEBUG] [paperless.consumer] [4ce923d5] Generating thumbnail for Test.pdf...

[2026-07-27 19:06:45,597] [DEBUG] [paperless.parsing] [4ce923d5] Execute: convert -density 300 -scale 500x5000> -alpha remove -strip -auto-orient -define pdf:use-cropbox=true /tmp/paperless/paperless-ngxi24ot2wy/Test.pdf[0] /tmp/paperless/paperless-tsw0dwwb/convert.webp

[2026-07-27 19:06:46,434] [INFO] [paperless.parsing] [4ce923d5] convert exited 0

[2026-07-27 19:06:49,222] [DEBUG] [paperless.consumer] [4ce923d5] Saving record to database

[2026-07-27 19:06:49,222] [DEBUG] [paperless.consumer] [4ce923d5] Creation date from st_mtime: 2026-07-27 19:06:45+00:00

[2026-07-27 19:06:50,205] [INFO] [paperless.handlers] [4ce923d5] Assigning document type Formular to 2026-07-27 Test

[ .... ] now, after setting Mode to force:

[2026-07-27 19:15:19,916] [DEBUG] [paperless.consumer] Archive: no — born-digital PDF (structure tags detected), ARCHIVE_FILE_GENERATION=auto

[2026-07-27 19:15:19,932] [INFO] [paperless.parsing.tesseract] pdftotext exited 0

[2026-07-27 19:15:19,934] [DEBUG] [paperless.parsing.tesseract] Text detection: original_has_text=True (text_length=0, mode=force, produce_archive=False)

[2026-07-27 19:15:19,934] [DEBUG] [paperless.parsing.tesseract] OCR strategy: full OCR — OCR_MODE=force

[2026-07-27 19:15:19,934] [DEBUG] [paperless.parsing.tesseract] Calling OCRmyPDF with args: {'input_file_or_options': PosixPath('/usr/src/paperless/media/documents/originals/0000207.pdf'), 'output_file': PosixPath('/tmp/paperless/paperless-v5w8og6f/archive.pdf'), 'use_threads': True, 'jobs': 8, 'language': 'deu+eng', 'output_type': 'pdfa', 'progress_bar': False, 'color_conversion_strategy': ColorConvertChoices.RGB, 'force_ocr': True, 'clean': True, 'deskew': True, 'rotate_pages': True, 'rotate_pages_threshold': 12.0, 'sidecar': PosixPath('/tmp/paperless/paperless-v5w8og6f/sidecar.txt')}

[2026-07-27 19:15:20,541] [WARNING] [ocrmypdf._pipeline] This PDF is marked as a Tagged PDF. This often indicates that the PDF was generated from an office document and does not need OCR. PDF pages processed by OCRmyPDF may not be tagged correctly.

[2026-07-27 19:15:21,448] [INFO] [ocrmypdf._exec.tesseract] [tesseract] Too few characters. Skipping this page

[2026-07-27 19:15:21,448] [INFO] [ocrmypdf._exec.tesseract] [tesseract] Too few characters. Skipping this page

[2026-07-27 19:15:21,448] [ERROR] [ocrmypdf._exec.tesseract] [tesseract] Error during processing.

[2026-07-27 19:15:21,448] [INFO] [ocrmypdf._pipeline] page is facing ⇧, confidence 0.00 - no change

[2026-07-27 19:15:23,906] [INFO] [ocrmypdf._graft] Parsing 1 pages with HocrParser

[2026-07-27 19:15:25,169] [INFO] [ocrmypdf._pipelines.ocr] Postprocessing...

[2026-07-27 19:15:25,557] [INFO] [ocrmypdf._pipeline] Image optimization ratio: 1.00 savings: 0.1%

[2026-07-27 19:15:25,558] [INFO] [ocrmypdf._pipeline] Total file size ratio: 0.18 savings: -447.2%

[2026-07-27 19:15:25,559] [INFO] [ocrmypdf._pipelines._common] Output file is a PDF/A-2b (as expected)
```

### Browser logs

```bash

```

### Services logs

```bash

```

### Paperless-ngx version

3.0.2

### Host OS

Ubuntu 24.04.4 LTS x86_64

### Installation method

Docker - official image

### System status

```json

```

### Browser

Firefox

### Configuration changes

_No response_

### Please confirm the following

- [x] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [x] This issue is not about the OCR or archive creation of a specific file(s). Otherwise, please see above regarding OCR tools.
- [x] I have already searched for relevant existing issues and discussions before opening this report.
- [x] I have updated the title field above with a concise description.

--- a/docs/configuration.md
+++ b/docs/configuration.md
@@ -932,8 +932,8 @@ for display in the web interface.
     | Document type              | `never` | `auto` (default)           | `always` |
     | -------------------------- | ------- | -------------------------- | -------- |
     | Scanned image (TIFF, JPEG) | No      | **Yes**                    | Yes      |
-    | Image-based PDF            | No      | **Yes** (short/no text, untagged) | Yes |
-    | Born-digital PDF           | No      | No (tagged or has embedded text)  | Yes |
+    | Image-based PDF            | No      | **Yes** (no embedded text) | Yes |
+    | Born-digital PDF           | No      | No (has embedded text, optionally confirmed by tag) | Yes |
     | Plain text, email, HTML    | No      | No                         | No       |
     | DOCX / ODT (via Tika)      | Yes\*   | Yes\*                      | Yes\*    |
 

--- a/src/documents/consumer.py
+++ b/src/documents/consumer.py
@@ -160,13 +160,14 @@ def should_produce_archive(
         _log.debug("Archive: yes — image document, ARCHIVE_FILE_GENERATION=auto")
         return True
     if mime_type == "application/pdf":
-        if is_tagged_pdf(document_path):
+        text = extract_pdf_text(document_path)
+        has_text = text is not None and len(text) > 0
+        if has_text and is_tagged_pdf(document_path):
             _log.debug(
                 "Archive: no — born-digital PDF (structure tags detected),"
                 " ARCHIVE_FILE_GENERATION=auto",
             )
             return False
-        text = extract_pdf_text(document_path)
         if text is None or len(text) <= PDF_TEXT_MIN_LENGTH:
             _log.debug(
                 "Archive: yes — scanned PDF (text_length=%d ≤ %d),"

--- a/src/documents/tests/test_consumer_archive.py
+++ b/src/documents/tests/test_consumer_archive.py
@@ -166,24 +166,28 @@ class TestShouldProduceArchive:
         mocker: MockerFixture,
         settings,
     ) -> None:
-        """Tagged PDFs (e.g. Word exports) are treated as born-digital regardless of text length."""
+        """Tagged PDFs (e.g. Word exports) with real text are treated as born-digital, even below PDF_TEXT_MIN_LENGTH."""
         settings.ARCHIVE_FILE_GENERATION = "auto"
         mocker.patch("documents.consumer.is_tagged_pdf", return_value=True)
+        mocker.patch("documents.consumer.extract_pdf_text", return_value="tiny")
         parser = _parser_instance(can_produce=True, requires_rendition=False)
         assert (
             should_produce_archive(parser, "application/pdf", Path("/tmp/doc.pdf"))
             is False
         )
 
-    def test_tagged_pdf_does_not_call_pdftotext(
+    def test_tagged_pdf_without_text_produces_archive(
         self,
         mocker: MockerFixture,
         settings,
     ) -> None:
-        """When a PDF is tagged, pdftotext is not invoked (fast path)."""
+        """A tagged PDF with no actual extractable text (e.g. some scanner firmware) is not
+        trusted as born-digital — the tag alone must not bypass OCR."""
         settings.ARCHIVE_FILE_GENERATION = "auto"
         mocker.patch("documents.consumer.is_tagged_pdf", return_value=True)
-        mock_extract = mocker.patch("documents.consumer.extract_pdf_text")
+        mocker.patch("documents.consumer.extract_pdf_text", return_value=None)
         parser = _parser_instance(can_produce=True, requires_rendition=False)
-        should_produce_archive(parser, "application/pdf", Path("/tmp/doc.pdf"))
-        mock_extract.assert_not_called()
+        assert (
+            should_produce_archive(parser, "application/pdf", Path("/tmp/doc.pdf"))
+            is True
+        )

--- a/src/paperless/parsers/tesseract.py
+++ b/src/paperless/parsers/tesseract.py
@@ -510,8 +510,10 @@ class RasterisedDocumentParser:
 
         if mime_type == "application/pdf":
             text_original = self.extract_text(None, document_path)
-            original_has_text = is_tagged_pdf(document_path, log=self.log) or (
-                text_original is not None and len(text_original) > PDF_TEXT_MIN_LENGTH
+            has_text = text_original is not None and len(text_original) > 0
+            original_has_text = has_text and (
+                is_tagged_pdf(document_path, log=self.log)
+                or len(text_original) > PDF_TEXT_MIN_LENGTH
             )
         else:
             text_original = None

--- a/src/paperless/tests/parsers/test_tesseract_parser.py
+++ b/src/paperless/tests/parsers/test_tesseract_parser.py
@@ -906,6 +906,34 @@ class TestSkipArchive:
         assert tesseract_parser.archive_path is None
         assert tesseract_parser.get_text()
 
+    def test_tagged_pdf_without_text_does_not_skip_ocr(
+        self,
+        mocker: MockerFixture,
+        tesseract_parser: RasterisedDocumentParser,
+        tesseract_samples_dir: Path,
+    ) -> None:
+        """
+        GIVEN:
+            - A PDF that reports itself as tagged (/MarkInfo /Marked true) but
+              has no actual extractable text (some scanner firmware produces
+              this — see GitHub issue #13349)
+            - Mode: auto, produce_archive=False
+        WHEN:
+            - Document is parsed
+        THEN:
+            - The tag alone is not trusted as "has text"; OCRmyPDF still runs
+        """
+        tesseract_parser.settings.mode = ModeChoices.AUTO
+        mocker.patch("paperless.parsers.tesseract.is_tagged_pdf", return_value=True)
+        mocker.patch.object(tesseract_parser, "extract_text", return_value=None)
+        mock_ocr = mocker.patch("ocrmypdf.ocr")
+        tesseract_parser.parse(
+            tesseract_samples_dir / "multi-page-images.pdf",
+            "application/pdf",
+            produce_archive=False,
+        )
+        mock_ocr.assert_called()
+
     def test_tagged_pdf_produces_pdfa_archive_without_ocr(
         self,
         tesseract_parser: RasterisedDocumentParser,

